source/solvers/postprocessing_velocities.cc

```diff
@@ -77,15 +74,14 @@ AverageVelocities<dim, VectorType, DofsType>::calculate_reynolds_stresses(
   else if constexpr (std::is_same_v<VectorType,
                                     TrilinosWrappers::MPI::BlockVector>)
     {
-      n_dofs_per_vertex = dim;
-      begin_index       = local_evaluation_point.block(0).local_range().first;
-      end_index         = local_evaluation_point.block(0).local_range().second;
-      local_solution    = &local_evaluation_point.block(0);
-      local_average     = &average_velocities.block(0);
-      rns_dt            = &reynolds_normal_stress_dt.block(0);
-      rss_dt            = &reynolds_shear_stress_dt.block(0);
-      k_dt              = &reynolds_normal_stress_dt.block(1);
-      k_index           = [](unsigned int i) { return i / dim; };
+      begin_index    = local_evaluation_point.block(0).local_range().first;
+      end_index      = local_evaluation_point.block(0).local_range().second;
+      local_solution = &local_evaluation_point.block(0);
+      local_average  = &average_velocities.block(0);
+      rns_dt         = &reynolds_normal_stress_dt.block(0);
+      rss_dt         = &reynolds_shear_stress_dt.block(0);
+      k_dt           = &reynolds_normal_stress_dt.block(1);
+      k_index        = [](unsigned int i) { return i / dim; };
```

Comment:
What is this initialization? Maybe add a comment to make it easier to understand :). I sense some fancy lambda shenanigans here.

Reply:
Ok! 
This is because for Trilinos vectors, (example with 3d, so (u,v,w,p)), k is going to be at the fourth position of the Reynolds normal stress vector (<u'u'>,<v'v'>,<v'v'>,k). Fourth position -> k_index = i + dim (i = first position of the uvwp package)
For Trinilos block vectors ([u,v,w][p]), k is going to be at the first position of the block 1. First position -> k_index = i / dim. (i = first position of the uvw package in block 0).
I use lambda because this is not just a coefficient that can help me to get the k position in its vector but 2 differents operator with dim and i. 
And I mean, why not use some C++ fancy stuff :p